templates/federated_reporting/mrproper.py

```diff
@@ -219,26 +294,94 @@ def main():
                 )
             )
             sys.exit(1)
-        print("Hosts to delete on {} are {}".format(hub["ui_name"], response["rows"]))
-
-        hosts_to_delete = response["rows"]
-        feederHostname = hub["ui_name"]
-        feederApi = NovaApi(
-            api_user="mrproper",
-            api_password=mrproper_password,
-            cert_path=CERT_PATH,
-            hostname=feederHostname,
+        logger.debug(
+            "Hosts to delete on {} are {}".format(hub["ui_name"], response["rows"])
         )
+        hosts_to_delete = response["rows"]
+        if len(hosts_to_delete) != 0:
+            logger.debug(
+                "{} host(s) to delete on feeder {}".format(
+                    len(hosts_to_delete), hub["ui_name"]
+                )
```

Comment:
Please use the printf-like replacements with the logging function. They do lazy evaluation and avoid constructing the string if it's not needed (discarded).